### ISSUE TITLE
CRN训练问题

### ISSUE THREAD

**Issue by lflyme** [seorim0/Speech_enhancement_with_Pytorch#1]
你好，请问你有做CRN的训练吗？我做CRN训练的时候发现数据如果不做归一化没法收敛，请问有遇到这种情况吗？或者CRN训练数据制作有哪些需要注意的，比如每个wav文件长度和归一化问题？

**Comment by ioyy900205** [comment]
你的没法收敛具体指什么？

**Comment by ioyy900205** [comment]
> 你好，请问你有做CRN的训练吗？我做CRN训练的时候发现数据如果不做归一化没法收敛，请问有遇到这种情况吗？或者CRN训练数据制作有哪些需要注意的，比如每个wav文件长度和归一化问题？

你的没法收敛 是 指 loss没法下降？

**Comment by seorim0** [comment]
> 你好，请问你有做CRN的训练吗？我做CRN训练的时候发现数据如果不做归一化没法收敛，请问有遇到这种情况吗？或者CRN训练数据制作有哪些需要注意的，比如每个wav文件长度和归一化问题？

First of all, please understand that there may be some misunderstandings because the question you asked was translated into English and then translated back to Korean.

Does the normalization you mentioned mean data normalization? Then, I did it like this:

For example, If the training data length is set to 3 seconds,
Clips shorter than 3 seconds are stretched to 3 seconds with zero padding, and clips longer than 3 seconds were cut to 3 seconds.
And the whole data was adjusted to values between -1 and 1 using a normalization term such as min-max.

**Comment by lflyme** [comment]
> > 你好，请问你有做CRN的训练吗？我做CRN训练的时候发现数据如果不做归一化没法收敛，请问有遇到这种情况吗？或者CRN训练数据制作有哪些需要注意的，比如每个wav文件长度和归一化问题？
> 
> 你的没法收敛 是 指 loss没法下降？

是的，不做归一化loss一直没法下降，做归一化才可以下降。请问你那边有遇到吗？

**Comment by yongjiankuang** [comment]
> 

yes，I did the training data normalization，then the training loss decrease. if i do not normalization the training data，the training loss do not decrease. Have you try to training with the data do not normalization，the training loss decrease step by step?

**Comment by yongjiankuang** [comment]
> 

yes，I did the training data normalization，then the training loss decrease. if i do not normalization the training data，the training loss do not decrease. Have you try to training with the data do not normalization，the training loss decrease step by step?

**Comment by ioyy900205** [comment]
> > > 你好，请问你有做CRN的训练吗？我做CRN训练的时候发现数据如果不做归一化没法收敛，请问有遇到这种情况吗？或者CRN训练数据制作有哪些需要注意的，比如每个wav文件长度和归一化问题？
> > 
> > 
> > 你的没法收敛 是 指 loss没法下降？
> 
> 是的，不做归一化loss一直没法下降，做归一化才可以下降。请问你那边有遇到吗？

I did not use normalization. The SI-SNR results using fullsubnet are stable between 10 and 11. This is about the same as the results of my own model.

**Comment by lflyme** [comment]
> > 你好，请问你有做CRN的训练吗？我做CRN训练的时候发现数据如果不做归一化没法收敛，请问有遇到这种情况吗？或者CRN训练数据制作有哪些需要注意的，比如每个wav文件长度和归一化问题？
> 
> First of all, please understand that there may be some misunderstandings because the question you asked was translated into English and then translated back to Korean.
> 
> Does the normalization you mentioned mean data normalization? Then, I did it like this:
> 
> For example, If the training data length is set to 3 seconds, Clips shorter than 3 seconds are stretched to 3 seconds with zero padding, and clips longer than 3 seconds were cut to 3 seconds. And the whole data was adjusted to values between -1 and 1 using a normalization term such as min-max.


yes，I did the training data normalization，then the training loss decrease. if i didn't normalization the training data，the training loss do not decrease. Have you try to training with the data do not normalization，the training loss decrease step by step?  I'm trying to training the model without data normalization，the lossfunction is mse，but the training loss much big，about 290000000

**Comment by andyye1999** [comment]
> > > 你好，请问你有做CRN的训练吗？我做CRN训练的时候发现数据如果不做归一化没法收敛，请问有遇到这种情况吗？或者CRN训练数据制作有哪些需要注意的，比如每个wav文件长度和归一化问题？
> > 
> > 
> > 你的没法收敛 是 指 loss没法下降？
> 
> 是的，不做归一化loss一直没法下降，做归一化才可以下降。请问你那边有遇到吗？

i have the same problems.  have you ever solved it ?
